pkg/cloud/cloud.go

```diff
@@ -88,3 +98,47 @@ func New(region, kmsEndpoint string, qps, burst, retryTokenCapacity int) (AWSKMS
 	client := kms.NewFromConfig(cfg, kmsOptFns...)
 	return client, nil
 }
+
+func addConfusedDeputyHeaders(cfg *aws.Config, sourceArn string) {
+	if sourceArn != "" {
+		sourceAccount, err := getSourceAccount(sourceArn)
+		if err != nil {
+			panic(fmt.Sprintf("%s is not a valid arn, err: %v", sourceArn, err))
```

Comment:
nit: let's just return error instead of panic to be consistent with what we have inside `New` function